extra_data/reader.py

```diff
@@ -1993,26 +1994,35 @@ def open_run(
         the proposal directory path. By default it looks for a file named
         ``{}/usr/extra-data-aliases.yml``.
     """
+    absence_ok = set()
+    if data == 'default':
+        data = ['proc', 'raw']
+        absence_ok = {'proc'}
     if data == 'all':
         data = ['raw', 'proc']
 
     if isinstance(data, Sequence) and not isinstance(data, str):
-        common_args = dict(
-            proposal=proposal, run=run, include=include,
-            file_filter=file_filter, inc_suspect_trains=inc_suspect_trains,
-            parallelize=parallelize)
+        base_dc = None
 
-        # Open the raw data
-        base_dc = open_run(**common_args, data=data[0], aliases=aliases)
-
-        for origin in data[1:]:
+        for origin in data:
             try:
                 # Attempt to open data at this origin, but this may not
                 # exist.
-                origin_dc = open_run(**common_args, data=origin, aliases=aliases)
+                origin_dc = open_run(
+                    proposal, run, data=origin, include=include,
+                    file_filter=file_filter, inc_suspect_trains=inc_suspect_trains,
+                    parallelize=parallelize, aliases=aliases,
+                )
             except FileNotFoundError:
-                warn(f'No data available for this run at origin {origin}')
+                if origin not in absence_ok:
+                    if base_dc is None:
+                        raise
+                    warn(f'No data available for this run at origin {origin}')
             else:
```

Comment:
I would probably go for a `continue` in the `except` branch, and move the entire `else` branch to after the complete `try-except` to avoid a level of indentation for so much code. I realize this has been like this before (written by me 🙃), but the logic seems complicated enough now that it seems useful.

Reply:
Good point, done